pgjdbc/src/test/java/org/postgresql/util/ReaderInputStreamTest.java

```diff
@@ -182,6 +184,31 @@ public void malformedSingleCharArrayReadTest() throws IOException {
     read(is, 0xF0, 0xA0, 0x9C, 0x8E);
   }
 
+  @Test
+  public void readsSmallerThanBlockSizeTest() throws Exception {
+    final int BLOCK = 8 * 1024;
+    final int DATASIZE = BLOCK * 5 + 57;
+    final byte[] data = new byte[DATASIZE];
+    final byte[] buffer = new byte[BLOCK];
+
+    // initialize with values (0, 1, ... 127, 0, 1, ...)
+    for (int i = 0; i < data.length; i++) {
+      data[i] = (byte) (i & 0x7F);
```

Comment:
It was used to check if the read blocks were contiguous (ie last block ended with value 33, next one should start with 34), but is not needed anymore (it only makes sense with custom charBufferSize in ReaderInputStream). Will remove it.